owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/sharing/OCShareViewModel.kt

```diff
@@ -0,0 +1,366 @@
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.owncloud.android.data.sharing.shares.db.OCShareEntity
+import com.owncloud.android.domain.sharing.shares.usecases.CreatePrivateShareUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.CreatePublicShareUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.DeleteShareUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.EditPrivateShareUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.EditPublicShareUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.GetShareAsLiveDataUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.GetSharesAsLiveDataUseCase
+import com.owncloud.android.domain.sharing.shares.usecases.RefreshSharesFromNetworkUseCase
+import com.owncloud.android.lib.resources.shares.ShareType
+import com.owncloud.android.operations.common.OperationType
+import com.owncloud.android.presentation.UIResult
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * View Model to keep a reference to the share repository and an up-to-date list of a shares
+ */
+class OCShareViewModel(
+    private val filePath: String,
+    val context: Context,
+    val account: Account,
+    getSharesAsLiveDataUseCase: GetSharesAsLiveDataUseCase = GetSharesAsLiveDataUseCase(context, account),
+    private val getShareAsLiveDataUseCase: GetShareAsLiveDataUseCase = GetShareAsLiveDataUseCase(context, account),
+    private val refreshSharesFromNetworkUseCase: RefreshSharesFromNetworkUseCase =
+        RefreshSharesFromNetworkUseCase(context, account),
+    private val createPrivateShareUseCase: CreatePrivateShareUseCase = CreatePrivateShareUseCase(context, account),
+    private val editPrivateShareUseCase: EditPrivateShareUseCase = EditPrivateShareUseCase(context, account),
+    private val createPublicShareUseCase: CreatePublicShareUseCase = CreatePublicShareUseCase(context, account),
+    private val editPublicShareUseCase: EditPublicShareUseCase = EditPublicShareUseCase(context, account),
+    private val deletePublicShareUseCase: DeleteShareUseCase = DeleteShareUseCase(context, account)
+) : ViewModel() {
+
+    private val _shares = MutableLiveData<UIResult<List<OCShareEntity>>>()
+    val shares: LiveData<UIResult<List<OCShareEntity>>> = _shares
+
+    private var sharesLiveData: LiveData<List<OCShareEntity>>? = getSharesAsLiveDataUseCase.execute(
+        GetSharesAsLiveDataUseCase.Params(
+            filePath = filePath,
+            accountName = account.name
+        )
+    ).data
+
+    // To detect changes in shares
+    private val sharesObserver: Observer<List<OCShareEntity>> = Observer { shares ->
+        _shares.postValue(UIResult.success(shares))
+    }
+
+    init {
+        sharesLiveData?.observeForever(sharesObserver)
```

Comment:
the same as before

Reply:
I answered in https://github.com/owncloud/android/pull/2702#discussion_r343020829